libcudacxx/include/cuda/std/source_location

```diff
@@ -0,0 +1,79 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX_SOURCE_LOCATION
+#define _LIBCUDACXX_SOURCE_LOCATION
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/cstdint>
+#include <cuda/std/version>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+//! NOTE: we cannot use `__builtin_source_location` as that requires us to use `::std::source_location::impl`
+class source_location
+{
+private:
+  const char* __file_name_     = "";
+  const char* __function_name_ = "";
+  uint_least32_t __line_       = 0;
+  uint_least32_t __column_     = 0;
```

Comment:
If there's any way to get away with it, any chance we stick with conventional names for these members (no double underscore)?

Reply:
Done